### PR TITLE
Added documentation for functions. Started creating calc_fitness

### PR DESCRIPTION
**Pull request by Marissa-Clark**
Still having trouble getting my pytests to be able to import my functions

### CODE REVIEW

File: geneticalg/examples/run_ga/run_ga.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import pandas as pd
+from geneticalg import gen_chromosome, calc_fitness_all, mutate
+
+video_data = pd.read_csv("./data/video_ts_data.csv", index_col=None)
+video_lengths = pd.read_csv("./data/video_lengths.csv").iloc[:, 0].values
+video_list = pd.read_csv("./data/video_list.csv",
```
Comment (yarikoptic) [inline geneticalg/examples/run_ga/run_ga.py:6]: you hardcoded the assumption that user (you) runs this script so these relative paths are correct.  (actually within the state of this branch I do not even see any `data/` directory).
To make it more robust and remove the assumption of where to run this script from, you can compose full path to the `./data` by taking the directory of this script: `os.path.dirname(__file__)`.

File: geneticalg/examples/run_ga/run_ga.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+# Generate Empty Chromosomes
+chromosomes = []
+
+for i in range(popSize):
+    chromosomes.append(gen_chromosome(max_seconds, video_list, video_lengths))
+
+calc_fitness_all(chromosomes)
+
+mutate(chromosomes[0], max_attempts, video_lengths, max_seconds)
```
Comment (yarikoptic) [inline geneticalg/examples/run_ga/run_ga.py:23]: don't you want also to demonstrate (print/display somehow) the result?

File: geneticalg/geneticalg.py [hunk 3]
```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/python
```
Comment (yarikoptic) [inline geneticalg/geneticalg.py:1]: generally it is advised to use `#!/usr/bin/env python` as shebang, so you use whatever python is currently "used" (according to `$PATH`) - might be system wide like this one or come from virtualenv, or from conda installation

File: geneticalg/geneticalg.py [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+def mutate(chromosome, max_attempts, video_lengths, max_seconds):
+    """Point mutation by removing one "gene" and adding another
+
+    Parameters
+    ----------
+    chromosome : np.ndarray
+        Dummy-coded chromosome of videos
+    max_attempts : int
+        Maximum number of tries to mutate (it may be possible theres no combo
+        that fits our criteria (< max_seconds))
```
Comment (yarikoptic) [inline geneticalg/geneticalg.py:112]: and what happens when max_attempts is reached?  I see that it then would quit the loop and return what `gen_chromosome()` would return (but that would not work since it requires arguments which aren't provided below)... so, imho something worth describing in the documentation here or better in the description above

File: geneticalg/tests/test_ga.py [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+
+def test_gen_chrom():
+    assert sum(gen_chrom(10, [1, 2], [1, 1])) <= 2
+    assert sum(gen_chrom(10, [1, 2], [5, 7])) < 2
+
+
+def test_calc_duration():
+    assert calc_duration([0, 0], np.array([1, 4])) == 0
+    assert calc_duration([0, 1], np.array([5, 5])) == 5
+    assert calc_duration([1, 1], np.array([2, 7])) == 9
```
Comment (yarikoptic) [inline geneticalg/tests/test_ga.py:13]: you have defined lovely functions above, which beg to be tested, especially for corner cases like the one noted for  `mutate`.

File: geneticalg/geneticalg.py (hunk 3 above)
Comment (yarikoptic) [reply geneticalg/geneticalg.py:1]: but should it even be a script (i.e. have shebang)?  I guess it doesn't hurt really, but since it is a module, could be omitted

**Review by yarikoptic — COMMENTED** [review]
Very nice, but even more tests would be great to spot e.g. incorrect code/logic.  You could even take your `examples/run_ga/run_ga.py` and make it into a test (if runs in reasonably quick time) or come up with some short(er) input data for it, but have the same "logic" and test outputs for viability.  Even without correctness/viability testing it would be useful to guarantee that the code at least runs (so called "smoke test")
